core/federated/RTI/main.c

```diff
@@ -316,9 +319,14 @@ int main(int argc, const char* argv[]) {
   assert(rti.base.number_of_scheduling_nodes < UINT16_MAX);
 
   // Allocate memory for the federates
-  rti.base.scheduling_nodes =
-      (scheduling_node_t**)calloc(rti.base.number_of_scheduling_nodes, sizeof(scheduling_node_t*));
-  for (uint16_t i = 0; i < rti.base.number_of_scheduling_nodes; i++) {
+  int n = rti.base.number_of_scheduling_nodes;
+  rti.base.scheduling_nodes = (scheduling_node_t**)calloc(n, sizeof(scheduling_node_t*));
+  // Allocate memory for the array of min_delays.
+  rti.base.min_delays = (tag_t*)calloc((n * n), sizeof(tag_t));
```

Comment:
Should this allocation occur only if `rti.base.dnet_enabled` is true?
The for loop below would also have to be made conditional.

Reply:
Not really. `min_delays` is where the delays between each federate are stored. For example, if there are two federate A and B with one connection, A -/10 ms/-> B, the matrix `min_delays` looks like the below.

I note that `scheduling_node_t` no more contains [delays (from upstream)](https://github.com/lf-lang/reactor-c/blob/7db4930deaa0e4100b017b1d274e24aecba11e0d/core/federated/RTI/rti_common.h#L64) but stores **ID**s of upstream or downstream only. Currently, if `A` and `B` are not connected directly, only `B` knows that `min_delay` from `A` is `10 ms` and this is used for computing `EIMT` of `B`. With this matrix, `A` can also know that the delay to `B` is `10 ms` and this is needed for computing `DNET` of `A`.
|                |       A          |       B     |
| ---------| ---------- | ------- |
| A             | NEVER      |   10 ms  |
| B             | FOREVER  |   NEVER|